src/index.ts

```diff
@@ -3,16 +3,30 @@ import { PullRequestEvent, PullRequestReviewEvent } from '@octokit/webhooks-type
 
 module.exports = (app: Probot) => {
   app.on(['pull_request.opened', 'pull_request.reopened', 'pull_request.labeled', 'pull_request.edited', 'pull_request_review'], async (context) => {
-    // reading configuration
-    const config: any = await context.config('autoapproval.yml')
-    context.log(config, '\n\nLoaded config')
     context.log('Repo: %s', context.payload.repository.full_name)
 
     const pr = context.payload.pull_request
     context.log('PR: %s', pr.html_url)
-    const prLabels: string[] = pr.labels.map((label: any) => label.name)
+    context.log('Action: %s', context.payload.action)
+
+    // NOTE(dabrady) When a PR is first opened, it can fire several different kinds of events if the author e.g. requests
+    // reviewers or adds labels during creation. This triggers parallel runs of our GitHub App, so we need to filter out
+    // those simultaneous events and focus just on the re/open event in this scenario.
+    //
+    // These simultaneous events contain the same pull request data in their payloads, and specify the 'updated at'
+    // timestamp to be the same as the 'created at' timestamp for the pull request. We can use this to distinguish events
+    // that are fired during creation from events fired later on.
+    if (!['opened', 'reopened'].includes(context.payload.action) && pr.created_at === pr.updated_at) {
+      context.log('Ignoring additional creation event: %s', context.payload.action)
+      return
+    }
```

Comment:
This is the bugfix. As outlined in the PR description, the other changes to this file are simply reordering log statements; I also made sure the action config isn't loaded if we're just going to ignore the event.

Reply:
This breaks now the logic when a single event `labeled` is raised. 

For example, when a user opened a PR without labels, but labels were added after the PR was created (can be the case in certain automation flows). Then, in this case, the created_at and updated_at will be the same and the PR will not be improved.

Can you please check this case?

Reply:
It seems it's the same for any other event coming after the PR was opened.

Reply:
Interesting 🤔 I tested that case several times, but using a human (myself) to add the label after opening the PR, and it worked just fine: the `updated_at` timestamp` was no longer the same as the `created_at` timestamp.

I wonder if, in the case of automation flows as you mentioned, the requests are so close together that the timestamps are the same? I will test this out myself, but can you also share your test evidence as well? 🙏🏻

Reply:
>I opened it without any required labels. Therefore PR is not approved. Then I added the required label. But the PR remained not approved.

@dkhmelenko I tested two different versions of that scenario just now, one in which there were no labels added during creation, and one in which there were. It works fine for me 🤔 Perhaps there is something else about your scenario that we're missing?

### Scenario 1
1. Open a PR with no labels (auto-approval requires a specific label)
2. Add required label
3. PR is auto-approved 

![Apr-06-2022 17-13-27](https://user-images.githubusercontent.com/2475919/162020197-6bb3aeb7-b8be-4dc5-9ca0-8839ef4af63f.gif)

### Scenario 2
1. Open a PR with an irrelevant label (auto-approval requires a specific label)
2. Add required label
3. PR is auto-approved

Reply:
Thanks for the detailed explanation! I just tried it once again and now it works. Apparently something went wrong that time. Now all good 👍

Reply:
Awesome 🙏🏻 Appreciate your help here.